File: backend/internal/domain/models/review.go
// backend/internal/domain/models/review.go

package models

import "time"

type Review struct {
    ID                int       `json:"id"`
    UserID            int       `json:"user_id"`
    EntityType        string    `json:"entity_type"`
    EntityID          int       `json:"entity_id"`
    Rating            int       `json:"rating"`
    Comment           string    `json:"comment,omitempty"`
    Pros              string    `json:"pros,omitempty"`
    Cons              string    `json:"cons,omitempty"`
    Photos            []string  `json:"photos,omitempty"`
    LikesCount        int       `json:"likes_count"`
    IsVerifiedPurchase bool     `json:"is_verified_purchase"`
    Status            string    `json:"status"`
    CreatedAt         time.Time `json:"created_at"`
    UpdatedAt         time.Time `json:"updated_at"`
	HelpfulVotes      int       `json:"helpful_votes"`       
	NotHelpfulVotes   int       `json:"not_helpful_votes"`   
    // поля для мультиязычности
    OriginalLanguage string                            `json:"original_language"`
    Translations    map[string]map[string]string      `json:"translations,omitempty"`
    
    // Дополнительные поля для отображения
    User              *User     `json:"user,omitempty"`
    Responses         []ReviewResponse `json:"responses,omitempty"`
    VotesCount        struct {
        Helpful     int `json:"helpful"`
        NotHelpful  int `json:"not_helpful"`
    } `json:"votes_count,omitempty"`
    CurrentUserVote   string    `json:"current_user_vote,omitempty"`
}

type ReviewResponse struct {
    ID        int       `json:"id"`
    ReviewID  int       `json:"review_id"`
    UserID    int       `json:"user_id"`
    Response  string    `json:"response"`
    CreatedAt time.Time `json:"created_at"`
    UpdatedAt time.Time `json:"updated_at"`
    User      *User     `json:"user,omitempty"`
}

type ReviewVote struct {
    ReviewID  int       `json:"review_id"`
    UserID    int       `json:"user_id"`
    VoteType  string    `json:"vote_type"`
    CreatedAt time.Time `json:"created_at"`
}

// Запрос на создание отзыва
type CreateReviewRequest struct {
    EntityType  string   `json:"entity_type" validate:"required,oneof=listing room car"`
    EntityID    int      `json:"entity_id" validate:"required"`
    Rating      int      `json:"rating" validate:"required,min=1,max=5"`
    Comment     string   `json:"comment"`
    Pros        string   `json:"pros"`
    Cons        string   `json:"cons"`
    Photos      []string `json:"photos"`
}

// Фильтры для получения отзывов
type ReviewsFilter struct {
    EntityType string `query:"entity_type"`
    EntityID   int    `query:"entity_id"`
    UserID     int    `query:"user_id"`
    MinRating  int    `query:"min_rating"`
    MaxRating  int    `query:"max_rating"`
    Status     string `query:"status"`
    SortBy     string `query:"sort_by"` // rating, date, likes
    SortOrder  string `query:"sort_order"` // asc, desc
    Page       int    `query:"page"`
    Limit      int    `query:"limit"`
}
type ReviewStats struct {
    TotalReviews      int     `json:"total_reviews"`
    AverageRating     float64 `json:"average_rating"`
    VerifiedReviews   int     `json:"verified_reviews"`
    RatingDistribution map[int]int `json:"rating_distribution"` // Распределение оценок: {1: 10, 2: 20, ...}
    PhotoReviews      int     `json:"photo_reviews"`          // Количество отзывов с фото
}

File: frontend/svetu/src/types/review.ts
export interface Review {
  id: number;
  user_id: number;
  entity_type: 'listing' | 'user' | 'storefront';
  entity_id: number;
  entity_origin_type?: string;
  entity_origin_id?: number;
  rating: number;
  comment?: string;
  pros?: string;
  cons?: string;
  photos?: string[];
  likes_count: number;
  is_verified_purchase: boolean;
  status: 'pending' | 'published' | 'deleted' | 'spam';
  created_at: string;
  updated_at: string;
  helpful_votes: number;
  not_helpful_votes: number;
  seller_confirmed: boolean;
  has_active_dispute: boolean;
  original_language: string;
  translations?: Record<string, Record<string, string>>;

  // Дополнительные поля для отображения
  user?: {
    id: number;
    name: string;
    avatar?: string;
  };
  responses?: ReviewResponse[];
  votes_count?: {
    helpful: number;
    not_helpful: number;
  };
  current_user_vote?: 'helpful' | 'not_helpful' | null;
}

export interface ReviewResponse {
  id: number;
  review_id: number;
  user_id: number;
  response: string;
  created_at: string;
  updated_at: string;
  user?: {
    id: number;
    name: string;
    avatar?: string;
  };
}

export interface CreateReviewRequest {
  entity_type: 'listing' | 'user' | 'storefront';
  entity_id: number;
  rating: number;
  storefront_id?: number;
  comment: string;
  pros?: string;
  cons?: string;
  photos?: string[];
  original_language: string;
}

export interface ReviewsFilter {
  entity_type?: string;
  entity_id?: number;
  user_id?: number;
  min_rating?: number;
  max_rating?: number;
  status?: string;
  sort_by?: 'rating' | 'date' | 'likes';
  sort_order?: 'asc' | 'desc';
  page?: number;
  limit?: number;
}

export interface ReviewStats {
  total_reviews: number;
  average_rating: number;
  verified_reviews: number;
  rating_distribution: Record<number, number>;
  photo_reviews: number;
}

export interface AggregatedRating {
  entity_id: number;
  entity_type: string;
  total_reviews: number;
  average: number;
  recent_rating: number;
  recent_reviews: number;
  recent_trend: 'up' | 'down' | 'stable';
  verified_percentage: number;
  last_review_at: string;
  distribution: Record<string, number>;
  breakdown: {
    listing_reviews?: number;
    storefront_reviews?: number;
    direct_reviews?: number;
  };
}

export interface CanReviewResponse {
  can_review: boolean;
  reason?: string;
  chat_activity?: {
    total_messages: number;
    buyer_messages: number;
    seller_messages: number;
    last_message_at: string;
    first_message_at: string;
  };
}

export interface ReviewConfirmation {
  review_id: number;
  seller_id: number;
  confirmed: boolean;
  confirmed_at?: string;
  notes?: string;
}

export interface ReviewDispute {
  id: number;
  review_id: number;
  user_id: number;
  reason: string;
  status: 'pending' | 'resolved' | 'rejected';
  admin_notes?: string;
  created_at: string;
  resolved_at?: string;
  resolved_by?: number;
}
